lib/route/route.js

```diff
@@ -679,6 +679,32 @@ const initRoutes = (pages) => {
   // serviceDataTimestamp = getTimestamp()
 }
 
+function asyncRoute (middleware) {
```

Comment:
I almost certain this should live in `route.js` - which is really about mapping url to routes and vice-versa. An entirely separate bit of middleware...   `use-async.js`?

That aside, could/should the asyncification be dealt with by the middleware itself?

Alternatively, could we pass `app` into it , initialise and then add a `useAsync` method to `app`?

Reply:
Have moved this into it's own `use-async.js` in middleware

RE: approach - there are a [few approaches](https://www.google.com/search?q=express+async+await) out there, some even monkey patch Express so you import the patching module, and `app.use` suddenly is able to handle a Promise correctly. Some simply export `use-async`, so very similar to what we've done. In the end I thought this approach strikes a sensible balance considering we don't need to use this approach in many places yet.

> should the asyncification be dealt with by the middleware itself

How'dya mean? If that means the entirety of each relevant middleware would have to wrap itself in a try catch, not sure that's better?

> , could we pass app into it , initialise and then add a useAsync method to app

Quick example?

Reply:
> should the asyncification be dealt with by the middleware itself

`app.use(useAsync(middleware.init))`

vs

`app.use(middleware)`
...
`middleare.init = () => useAsync(initMethod)
```

> could we pass app into it , initialise and then add a useAsync method to app

literally just stick it on the app

`app.useAsync`

```
useAsync = (middleware) => {
  let wrappedMiddleware = ...
  app.use(wrappedMiddleware)
}
```

Reply:
Happy to stick with `app.use(useAsync(middleware))`, especially as we haven't got that many instances of it at the moment. We can always revisit…